lib/Entity/Layout.php

```diff
@@ -1436,11 +1447,41 @@ public function toZip($dataSetFactory, $fileName, $options = [])
                     $dataSetIds[] = $dataSet->dataSetId;
                     $dataSets[] = $dataSet;
                 }
+            }   elseif ($widget->type == 'subplaylist') {
+                $playlistIds = json_decode($widget->getOptionValue('subPlaylistIds', []), true);
+
+                foreach ($playlistIds as $playlistId) {
+                    $count = 1;
+                    $playlist = $this->playlistFactory->getById($playlistId);
+                    $playlist->load();
+                    $playlist->expandWidgets(0, false);
+                    $playlistDefinitions[$playlist->playlistId] = $playlist;
+
+                    // we are adding Playlist definitions, Playlist mappings and DataSets existing on the nested Playlist.
+                    $playlistMappings = $playlist->generatePlaylistMapping($playlist->widgets, $playlist->playlistId,$playlistMappings, $count, $nestedPlaylistDefinitions, $dataSetIds, $dataSets, $dataSetFactory, $options['includeData']);
```

Comment:
We also pass `$playlistMappings` by reference, so we do need to assign it as well?